common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -43,19 +43,33 @@ case class MDC(key: LogKey, value: Any)
  * Wrapper class for log messages that include a logging context.
  * This is used as the return type of the string interpolator `LogStringContext`.
  */
-case class MessageWithContext(message: String, context: java.util.HashMap[String, String]) {
+case class MessageWithContext(
+    message: String,
+    context: java.util.HashMap[String, String],
+    stripMarginFromMessage: Boolean = false) {
   def +(mdc: MessageWithContext): MessageWithContext = {
     val resultMap = new java.util.HashMap(context)
     resultMap.putAll(mdc.context)
-    MessageWithContext(message + mdc.message, resultMap)
+    MessageWithContext(
+      message + mdc.message,
+      resultMap,
+      mdc.stripMarginFromMessage || stripMarginFromMessage)
   }
+
+  def stripMargin: MessageWithContext = copy(stripMarginFromMessage = true)
```

Comment:
We can also have
```
def stripMargin: MessageWithContext = copy(message = message.stripMargin)
```
so that we don't need a boolean flag stripMarginFromMessage

Reply:
Good idea, this is done.